compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -137,32 +137,160 @@
+      {
+      return node->getByte();
+      }
+
+   static int16_t getNodeShortValue(TR::Node * node)
+      {
+      return node->getShortInt();
+      }
+
+   static int32_t getNodeIntValue(TR::Node * node)
+      {
+      return node->getInt();
+      }
+
+   static int64_t getNodeLongValue(TR::Node * node)
+      {
+      return node->getLongInt();
+      }
+
+   static uint32_t getNodeFloatBits(TR::Node * node)
+      {
+      return node->getFloatBits();
+      }
+
+   static uint64_t getNodeDoubleBits(TR::Node * node)
+      {
+      return node->getDoubleBits();
+      }
+
+   static void setNodeByteValue(TR::Node * node, int8_t value)
+      {
+      node->setByte(value);
+      }
+
+   static void setNodeShortValue(TR::Node * node, int16_t value)
+      {
+      node->setShortInt(value);
+      }
+
+   static void setNodeIntValue(TR::Node * node, int32_t value)
+      {
+      node->setInt(value);
+      }
+
+   static void setNodeLongValue(TR::Node * node, int64_t value)
+      {
+      node->setLongInt(value);
+      }
+
+   static void setNodeFloatBits(TR::Node * node, uint32_t bits)
+      {
+      node->setFloatBits(bits);
+      }
+
+   static void setNodeDoubleBits(TR::Node * node, uint64_t bits)
+      {
+      node->setDoubleBits(bits);
+      }
+   };
+
+template <typename T>
+struct BinaryOpSimplifier
+   {
+   TR::Simplifier * s;
+   T (* getNodeValue)(TR::Node *);
+   void (* setNodeValue)(TR::Node *, T);
+
+   /**
+   * Binary identity operation
+   *
+   * If the second child is a constant that represents an identity operation,
+   * replace this node with the first child.
+   */
+   inline TR::Node * tryAndSimplifyIdentityOp(TR::Node * node, T nullValue)
+      {
+      auto secondChild = node->getSecondChild();
+      if (!secondChild || !secondChild->getOpCode().isLoadConst())
+         return NULL;
+
+      if (getNodeValue(secondChild) != nullValue)
+         return NULL;
+
+      return s->replaceNode(node, node->getFirstChild(), s->_curTree);
+      }
+
+   /**
+   * Binary identity or zero operation
+   *
+   * If the second child is a constant that represents an identity operation,
+   * replace this node with the first child.
+   * Binary zero operation
+   *
+   * If the second child is a constant that represents a zero operation,
+   * replace this node with the second child.
+   */
+   inline TR::Node * tryAndSimplifyIdentityOrZeroOp(TR::Block * block, TR::Node * node, T nullValue, T zeroValue)
+      {
+      auto secondChild = node->getSecondChild();
+      if (!secondChild || !secondChild->getOpCode().isLoadConst())
+         return NULL;
+
+      auto value = getNodeValue(secondChild);
+
+      if (value == nullValue)
+         return s->replaceNodeWithChild(node, node->getFirstChild(), s->_curTree, block);
+
+      if (value != zeroValue)
+         return NULL;
+
+      if (performTransformation(s->comp(), "%sFound op with iconst in node [" POINTER_PRINTF_FORMAT "]\n", s->optDetailString(),node))
```

Comment:
Sorry - I have so many comments on code that wasn't changed.  In this case, the operand might not be an `iconst`.  Perhaps you could change this to use `node->getOpCode().getName()` to get the actual name of the opcode?